vllm/entrypoints/openai/serving_chat.py

```diff
@@ -507,6 +513,31 @@ def extract_tool_call_required_streaming(
 
         return delta_message, function_name_returned
 
+    def maybe_filter_parallel_tool_calls(
+        self, choice: ChatCompletionResponseChoiceT, request: ChatCompletionRequest
+    ) -> ChatCompletionResponseChoiceT:
+        """Filter to first tool call only when parallel_tool_calls is False."""
```

Comment:
Thanks, I think this is a reasonable change. It implements the semantics when `parallel_tool_calls` is set to False.

Could you move this function to the `vllm/entrypoints/chat_utils.py` file?